modules/validations/duplicate_address.js

```diff
@@ -0,0 +1,131 @@
+import { t } from '../util/locale';
+import { utilArrayIntersection, utilDisplayLabel } from '../util';
+import { validationIssue } from '../core/validation';
+
+
+export function validationDuplicateAddress(context) {
+  const type = 'duplicate_address';
+
+
+  function hasAddress(entity) {
+    return Object.keys(entity.tags).some(k => /^addr:/.test(k));
+  }
+
+
+  function isDuplicate(entity1, entity2) {
+    if (entity1.id === entity2.id) return false;   // ignore self
+
+    let dupe = { entity1: entity1, entity2: entity2, tags: {} };
+    let different = false;
+
+    // if any of these tags don't match, *and both are present*, it's not a duplicate..
+    ['addr:street', 'addr:place', 'addr:block'].forEach(k => {
+      const val1 = entity1.tags[k] || '';
+      const val2 = entity2.tags[k] || '';
+      if (val1 && val2) {     // both assigned
+        if (val1 !== val2) {
+          different = true;   // values don't match
+        } else {
+          dupe.tags[k] = val1;
+        }
+      }
+    });
+    if (different) return false;
+
+    // if any of these tags don't match, *and one is present*, it's not a duplicate..
+    ['addr:door', 'addr:unit', 'addr:flats', 'addr:floor'].forEach(k => {
+      const val1 = entity1.tags[k] || '';
+      const val2 = entity2.tags[k] || '';
+      if (val1 || val2) {     // one is assigned
+        if (val1 !== val2) {
+          different = true;   // values don't match
+        } else {
+          dupe.tags[k] = val1;
+        }
+      }
+    });
+    if (different) return false;
+
+    // compare housenumbers, which may contain multiple values
+    const housenumbers1 = (entity1.tags['addr:housenumber'] || '').split(/[;,]/).filter(Boolean);
```

Comment:
For my own information: Is one delimiter more favorable than the other (semicolon vs comma)?

Reply:
Semicolon is preferred/documented, but comma can also found in the wild.  
H/T @taylorsmock for the advice on this!  🎩

Reply:
Both are technically used. Last time I read the address wiki page, it indicated that either one could be used. I prefer `;`, since that is the standard delimiter for most other items (example, `turn:lanes=left;through|through;right` -- `|` delimits lanes, `;` delimits values, another example would be destinations, `destination=Washington, D.C.;Portland, Oregon`).